dotnet_task_memory_usage/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"runtime"
+	"strings"
+	"sync/atomic"
+	"time"
+)
+
+func main() {
+	numTasks := flag.Int("numtasks", 1, "number of tasks to run")
+	sleep := flag.Duration("sleep", 1*time.Second, "duration to sleep")
+	flag.Parse() // updates the pointers above to whatever values are provided
+	log.Printf("numTasks=%d sleep=%s\n", *numTasks, *sleep)
+
+	log.Printf("OS threads for Goroutine worker pool: %d", runtime.GOMAXPROCS(0))
+
+	scoreboard := Scoreboard{lastDump: time.Now()}
+
+	for i := 0; i < *numTasks; i++ {
+		go workTask(&scoreboard, *sleep)
+	}
+
+	scoreboard.PollScores()
+}
+
+func workTask(scoreboard *Scoreboard, sleep time.Duration) {
+	for {
+		time.Sleep(sleep)
+		scoreboard.AddHit()
+	}
+}
+
+type Scoreboard struct {
+	lastDump time.Time
+	score    int64
+}
+
+func (s *Scoreboard) AddHit() {
+	atomic.AddInt64(&s.score, 1)
+}
+
+func (s *Scoreboard) PollScores() {
+	for i := 0; i < 10; i++ {
+		time.Sleep(1 * time.Second)
+		totalHits := atomic.LoadInt64(&s.score)
+		elapsed := time.Since(s.lastDump)
+		s.lastDump = time.Now()
```

Comment:
I think there is a race here, you appear to be calling current time twice, so time could be "lost".
The C# version makes sure to call current time once and swaps the variable.
I doubt it is affecting the test results, but we should rule it out.